reporter.py

```diff
@@ -48,19 +49,25 @@ def raise_error_on_not_ok(response: Response, message: str):
 
 
 def report(nickname: str, challenge_name: str, run_id: str, result: Union[ChallengeResult, ChallengeError]):
+    if NO_UPLOAD:
```

Comment:
instead of this plz add another reporter module like `local_reporter`.
based on the value of the env variable decide which one should be  imported/injected.